linera-chain/src/data_types.rs

```diff
@@ -270,13 +270,16 @@ pub struct ExecutedBlock {
 
 /// The messages and the state hash resulting from a [`Block`]'s execution.
 #[derive(Debug, PartialEq, Eq, Hash, Clone, Serialize, Deserialize, SimpleObject)]
+#[cfg_attr(with_testing, derive(Default))]
 pub struct BlockExecutionOutcome {
     pub messages: Vec<OutgoingMessage>,
     /// For each transaction, the cumulative number of messages created by this and all previous
     /// transactions, i.e. `message_counts[i]` is the index of the first message created by
     /// transaction `i + 1` or later.
     pub message_counts: Vec<u32>,
     pub state_hash: CryptoHash,
+    /// The record of oracle responses for each transaction.
+    pub oracle_records: Vec<OracleRecord>,
```

Comment:
To be consistent, should we use `oracle_responses: Vec<OracleResponse>` and (maybe) `oracle_response_counts: Vec<u32>` ?

Reply:
Ah, I see! Yes, we could do that, too.

On the other hand, the motivation for the message counts was that we want to be able to address both the _block_'s `n`-th message and a particular _transaction_'s `n`-th message. I don't think that's the case for oracles.

Reply:
Then it's even better

Reply:
Not sure what you mean: Which variant do you prefer?

Reply:
`oracle_responses: Vec<OracleResponse>`